src/key.js

```diff
@@ -826,6 +826,30 @@ Key.prototype.verifyAllUsers = async function(keys) {
   return results;
 };
 
+Key.prototype.addSubkey = async function(options = {}) {
+  if (!this.isPrivate()) {
+    throw new Error("Cannot add a subkey to a public key");
+  }
+  const defaultOptions = {};
+  const defaultAlgo = this.primaryKey.algorithm;
+  if (defaultAlgo.indexOf('rsa') === 0) {
+    defaultOptions.numBits = this.primaryKey.getAlgorithmInfo().bits;
+  } else if (defaultAlgo === 'ecdh' || defaultAlgo === 'ecdsa' || defaultAlgo === 'eddsa') {
+    defaultOptions.curve = this.primaryKey.getAlgorithmInfo().curve;
+  }
```

Comment:
It seems like all of this could be simplified to
```js
const defaultOptions = this.primaryKey.getAlgorithmInfo();
defaultOptions.numBits = defaultOptions.bits;
```
Or, we could even remove the last line if we rename `numBits` to `bits` everywhere to make it consistent within the library, and then do something like
```js
export function generateKey({ userIds = [], passphrase = "", numBits = 2048, bits = numBits, keyExpirationTime = 0, curve = "", date = new Date(), subkeys = [{}] }) {
```
in src/openpgp.js (to keep it backwards compatible). But maybe that's something for another day. Up to you :)

Reply:
Updated